RealmSwift/RealmProperty.swift

```diff
@@ -0,0 +1,85 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+
+import Foundation
+import Realm
+
+/**
+ A `RealmProperty` instance represents an polymorphic value for supported types.
+
+ To change the underlying value stored by a `RealmProperty` instance, mutate the instance's `value` property.
+
+ - Note:
+ An `RealmProperty` should not be declared as `@objc dynamic` on a Realm Object. Use `let` instead.
+ */
+public final class RealmProperty<Value: RealmPropertyType>: RLMValueBase {
```

Comment:
Is there any reason this is generic? Seems like this is only useful for `AnyRealmValue`

Reply:
It will pretty much replace RealmOptional, for the short while that property wrappers aren't around. @tgoyne do you have any opinions on restricting this to just `AnyRealmValue`?